### PR TITLE
Add 3.20 noetic

### PR DESCRIPTION
**Pull request by at-wat**
## Package version updates
- pcl: `1.13.0` -> `1.14.1`
- boost: `1.80.0` -> `1.84.0`
- gcc: `12.2.1_git20220924-r4` -> `13.2.1_git20240309-r0`

## Major changes to be patched
- `NULL` is changed from `(int)0` to `()nullptr`
  - `boost::function` constructor can receive `(int)0` to initialize as empty (not `nullptr`)
- Python 3.12 dropped deprecated stdlib packages and functions
  - stdlib `imp` is removed and should be rewritten to use `importlib`
  - `unittest` deprecated assert functions are removed https://docs.python.org/3.11/library/unittest.html#deprecated-aliases
- Python 3.11 changed behavor
  - Caller stack frame seems be changed (related to `rosgraph`)
- PCL removed ROS message compatibility of `pcl::PointCloud`
  - `ifm3d` used this
- Boost cleaned implicit header includes
  - Some implicitly included headers aren't included in this version and require explicit include.
- Boost dropped deprecated headers
  - `boost/progress.hpp` is removed and `progress_display` is available under `boost/timer/progress_display.hpp`
- Nose dropped deprecated features
  - `assert_regexp_matches` is removed

## Note
- Gcc segmentation fault problem still exists

### CODE REVIEW

File: build-repo.sh [hunk 1]
```diff
@@ -34,6 +37,10 @@ case "${STACK_PROTECTOR:-yes}" in
 esac
 echo "export CFLAGS=\"${CFLAGS}\"" | sudo tee -a /etc/abuild.conf
 
+if [ "${ALPINE_VERSION}" != "3.17" ]; then
+  echo "export CXXFLAGS=\"-std=c++14\"" | sudo tee -a /etc/abuild.conf
```
Comment (at-wat) [inline build-repo.sh:41]: Gtest version on Alpine 3.20 requires C++14 or later

**Comment by at-wat** [comment]
Triggering clean rebuild

**Review by f-fl0 — APPROVED** [review]
Great! Thank you for going through all the packages and patching so many issues.

**Review by Taka-Kazu — APPROVED** [review]

**Comment by at-wat** [comment]
```
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
~~  traversing 1 packages in topological order:
~~  - self_test
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
The packages or cmake arguments have changed, forcing cmake invocation

==> Processing catkin package: 'self_test'
==> Creating build directory: 'build_isolated/self_test'
==> cmake /home/builder/aports/noetic/ros-noetic-self-test/abuild/src/self_test -DCATKIN_DEVEL_PREFIX=/home/builder/aports/noetic/ros-noetic-self-test/abuild/devel_isolated/self_test -DCMAKE_INSTALL_PREFIX=/home/builder/aports/noetic/ros-noetic-self-test/abuild/install_isolated -DCMAKE_BUILD_TYPE=RelWithDebInfo -G Unix Makefiles in '/home/builder/aports/noetic/ros-noetic-self-test/abuild/build_isolated/self_test'
CMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


-- The C compiler identification is GNU 13.2.1
-- The CXX compiler identification is GNU 13.2.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/lib/ccache/bin/gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/lib/ccache/bin/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Using CATKIN_DEVEL_PREFIX: /home/builder/aports/noetic/ros-noetic-self-test/abuild/devel_isolated/self_test
-- Using CMAKE_PREFIX_PATH: /usr/ros/noetic
-- This workspace overlays: /usr/ros/noetic
CMake Warning (dev) at /usr/ros/noetic/share/catkin/cmake/python.cmake:4 (find_package):
  Policy CMP0148 is not set: The FindPythonInterp and FindPythonLibs modules
  are removed.  Run "cmake --help-policy CMP0148" for policy details.  Use
  the cmake_policy command to set the policy and suppress this warning.

Call Stack (most recent call first):
  /usr/ros/noetic/share/catkin/cmake/all.cmake:164 (include)
  /usr/ros/noetic/share/catkin/cmake/catkinConfig.cmake:20 (include)
  CMakeLists.txt:6 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Found PythonInterp: /usr/bin/python3 (found suitable version "3.12.3", minimum required is "3")
-- Using PYTHON_EXECUTABLE: /usr/bin/python3
-- Using default Python package layout
-- Found PY_em: /usr/lib/python3.12/site-packages/em.py
-- Using empy: /usr/lib/python3.12/site-packages/em.py
-- Using CATKIN_ENABLE_TESTING: ON
-- Call enable_testing()
-- Using CATKIN_TEST_RESULTS_DIR: /home/builder/aports/noetic/ros-noetic-self-test/abuild/build_isolated/self_test/test_results
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- Found PkgConfig: /usr/bin/pkg-config (found version "2.2.0")
-- Found GTest: /usr/lib/cmake/GTest/GTestConfig.cmake (found version "1.14.0")
-- Checking for module 'gtest'
--   Found gtest, version 1.14.0
-- Using GTest v1.14.0 (parsed from pkg-config)
-- Found gmock: gmock and gtests will be built
-- Using Python nosetests: /usr/bin/nosetests
-- catkin 0.8.10
-- BUILD_SHARED_LIBS is on
-- Found Boost: /usr/lib/cmake/Boost-1.84.0/BoostConfig.cmake (found version "1.84.0") found components: thread
-- Configuring done (1.2s)
-- Generating done (0.1s)
-- Build files have been written to: /home/builder/aports/noetic/ros-noetic-self-test/abuild/build_isolated/self_test
==> make in '/home/builder/aports/noetic/ros-noetic-self-test/abuild/build_isolated/self_test'
<== Finished processing package [1 of 1]: 'self_test'
Built target tests
Removing test result files from '/home/builder/aports/noetic/ros-noetic-self-test/abuild/build_isolated/self_test/test_results/self_test'
Built target clean_test_results_self_test
the rosdep view is empty: call 'sudo rosdep init' and 'rosdep update'
the rosdep view is empty: call 'sudo rosdep init' and 'rosdep update'
the rosdep view is empty: call 'sudo rosdep init' and 'rosdep update'
the rosdep view is empty: call 'sudo rosdep init' and 'rosdep update'
�[33m[ WARN] [1717385120.038062000]: Non-zero self-test test status. Name: 'Error test', status 2: 'Reporting an error'�[0m
�[33m[ WARN] [1717385120.040788000]: Non-zero self-test test status. Name: 'Exception generating test', status 2: 'Uncaught exception: we did something that threw an exception'�[0m
�[33m[ WARN] [1717385120.042680000]: setID was not called by any self-test. The node author should be notified. If there is no suitable ID for this node, an ID of 'none' should be used.�[0m
... logging to /home/builder/aports/noetic/ros-noetic-self-test/abuild/log/rostest-f99e8e472ac9-176882.log
[ROSUNIT] Outputting test results to /home/builder/aports/noetic/ros-noetic-self-test/abuild/build_isolated/self_test/test_results/self_test/rostest-test_launch_nominal_test.xml
[Testcase: testtest_self_test_nominal] ... ok

[ROSTEST]-----------------------------------------------------------------------

[self_test.rosunit-test_self_test_nominal/test_self_test][ERROR]----------------
'TestSelfTest' object has no attribute 'assert_'
  File "/usr/lib/python3.12/unittest/case.py", line 58, in testPartExecutor
    yield
  File "/usr/lib/python3.12/unittest/case.py", line 634, in run
    self._callTestMethod(testMethod)
  File "/usr/lib/python3.12/unittest/case.py", line 589, in _callTestMethod
    if method() is not None:
       ^^^^^^^^
  File "/home/builder/aports/noetic/ros-noetic-self-test/abuild/src/self_test/scripts/test_selftest.py", line 91, in test_self_test
    self.assert_(res.id == '', "Result had node ID even though ID was not expected. ID: %s" % res.id)
    ^^^^^^^^^^^^
--------------------------------------------------------------------------------


SUMMARY
�[1;31m * RESULT: FAIL�[0m
 * TESTS: 1
�[1;31m * ERRORS: 1�[0m
 * FAILURES: 0
```
not sure why this doesn't failed on PR

**Pull request merged**